tests/rptest/tests/fetch_fairness_test.py

```diff
@@ -0,0 +1,103 @@
+from rptest.services.rpk_producer import RpkProducer
+from rptest.services.kaf_producer import KafProducer
+from rptest.services.admin import Admin
+
+
+class FetchTest(RedpandaTest):
+    def __init__(self, test_ctx, *args, **kwargs):
+        self._ctx = test_ctx
+        super(FetchTest, self).__init__(test_ctx,
+                                        num_brokers=1,
+                                        *args,
+                                        extra_rp_conf={},
+                                        **kwargs)
+
+    """
+    Validates key availability properties of the system using a single
+    partition.
+    """
+
+    @cluster(num_nodes=1)
+    @parametrize(type='multiple-topics')
+    @parametrize(type='multiple-partitions')
+    def simple_fetch_handler_fairness_test(self, type):
+        """
+        Testing if when fetching from single node all partitions are 
+        returned in round robin fashion
+        """
+        def multiple_topics(count):
+            return list(
+                map(
+                    lambda i: TopicSpec(partition_count=1,
+                                        replication_factor=1), range(0,
+                                                                     count)))
+
+        def multiple_partitions(count):
+            return [TopicSpec(partition_count=count, replication_factor=1)]
+
+        number_of_partitions = 32
+        records_per_partition = 10
+        topics = []
+
+        if type == 'multiple-topics':
+            topics = multiple_topics(number_of_partitions)
+        else:
+            topics = multiple_partitions(number_of_partitions)
+
+        # create topics
+        self.redpanda.create_topic(specs=topics)
+        self.redpanda.logger.info(f"topics: {topics}")
+        rpk = RpkTool(self.redpanda)
+
+        # publish 10 messages to each topic/partition
+
+        for s in topics:
+            t = s.name
+            for p in range(0, s.partition_count):
+                for i in range(0, records_per_partition):
+                    self.redpanda.logger.info(f"producing to : {t}/{p}")
+                    rpk.produce(t,
+                                f"k.{t}.{p}.{i}",
+                                f"p.{t}.{p}.{i}",
+                                partition=p)
+
+        # configure kcl to fetch at most 1 byte in single fetch request,
+        # this way we should receive exactly one record per each partition
+
+        kcl = KCL(self.redpanda)
+        consumed = kcl.consume(topic="topic-.*",
+                               n=number_of_partitions,
+                               regex=True,
+                               fetch_max_bytes=1,
+                               group="test-gr-1")
+        consumed_partitions = set()
+        for c in consumed.split():
+            parts = c.split('.')
+            consumed_partitions.add((parts[1], parts[2]))
+
+        assert len(consumed_partitions) == number_of_partitions
```

Comment:
Same question as Alex: how does this validate the property of fetch session fairness?

Related: I think that asserting a particular order might result in a flaky test because it won't be ever be tolerant to any sort of faults, rebalances, etc... that we might want to add or inject in the future. One way to avoid this is to perform lots of little fetches like you've done, but then instead of verifying an order, verify that each partition was fetched from with an equal frequency as the other partitions (subject to a small threshold value).

Reply:
Agreed with Noah.

Reply:
the test i wrote does exactly what @dotnwat  suggested. I setup `max-fetch-bytes` in `kcl consume` this forces redpanda to return single message in `FetchResposne`. Requesting 64 messages with `-n` parameter of kcl results in a series of 64 `FetchRequests` sent to redpanda. Set is used to measure frequency of messages - expected frequency of each partition is 1. In order for set to contain `n` elements each has to be unique. 

Note on order:
Client shuffles order of partitions in `FetchRequest` hence checking ordering has no point in this test.

Reply:
got it. did you need to add any sort of error threshold to the balancing?

Reply:
i mean, i dont' think you do but if leadership transfer occurred, for example, perfect balance wouldn't occur

Reply:
i setup single node test to make test deterministic